### PR TITLE
[GH-15759] Fix bot username in RHS

### PR DESCRIPTION
**Pull request by marcelo-cardozo**
#### Summary

This pull request fixes the issue found when replying a message and typing, in the RHS commands executed by a Bot.
Eg: `/github help`

The issue was happening because the RHS treated the Bot Message as a System Message and was overriding the username for System. 
#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/15759
Jira Ticket https://mattermost.atlassian.net/browse/MM-29266

#### Screenshots

Screenshot from 2020-10-18 10-33-21

### CODE REVIEW

File: components/rhs_comment/rhs_comment.jsx [hunk 1]
```diff
@@ -361,6 +368,22 @@ class RhsComment extends React.PureComponent {
...
+                    />
+                );
+
+                visibleMessage = (
+                    <span className='post__visibility'>
+                        <FormattedMessage
+                            id='post_info.message.visible'
+                            defaultMessage='(Only visible to you)'
+                        />
+                    </span>
```
Comment (hanzei) [inline components/rhs_comment/rhs_comment.jsx:385]: Is this shown to only ephemeral bot posts? Or to all?

File: components/rhs_comment/rhs_comment.jsx [hunk 2]
```diff
@@ -132,8 +137,10 @@ class RhsComment extends React.PureComponent {
 
     handleShortcutReactToLastPost = (isLastPost) => {
         if (isLastPost) {
-            const {isReadOnly, channelIsArchived, enableEmojiPicker, post,
-                actions: {emitShortcutReactToLastPostFrom}} = this.props;
+            const {
+                isReadOnly, channelIsArchived, enableEmojiPicker, post,
+                actions: {emitShortcutReactToLastPostFrom},
+            } = this.props;
```
Comment (hanzei) [inline components/rhs_comment/rhs_comment.jsx:143]: That seems like an unrelated change. Does it have to be in this PR?

**Review by hanzei — CHANGES_REQUESTED** [review]
Thanks for the quick fix :+1: One question about non-ephemeral bot posts.

File: components/rhs_comment/rhs_comment.jsx (hunk 2 above)
Comment (marcelo-cardozo) [reply components/rhs_comment/rhs_comment.jsx:143]: Ohh, that was the autoformatting of my IDE, I will update it

File: components/rhs_comment/rhs_comment.jsx (hunk 1 above)
Comment (marcelo-cardozo) [reply components/rhs_comment/rhs_comment.jsx:385]: It is shown to all bot posts

**Review by larkox — APPROVED** [review]
LGTM. Thanks! :tada:

**Review by hanzei — APPROVED** [review]
LGTM :+1:

**Comment by amyblais** [comment]
@DHaussermann Kind reminder to review for v5.29.

**Review by DHaussermann — APPROVED** [review]
Tested and passed.
- This issue is resolved
- Bot ephemeral posts now display as expected when viewing them in the RHS
- Briefly regression tested no issues found 

LGTM! 

Thanks @marcelo-cardozo for this enhancement! much appreciated.

I have added test coverage for this to TM4J for release testing.
